kernel/data-pipeline/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/core/importer/connector/CDCImporterConnector.java

```diff
@@ -181,38 +208,62 @@ private final class CDCIncrementalImporterTask implements Runnable {
         @Override
         public void run() {
             while (running && null != dataRecordComparator) {
-                int index = 0;
-                List<Record> dataRecords = new LinkedList<>();
-                Map<CDCImporter, Record> pipelineChannelPositions = new HashMap<>(incrementalRecordMap.size(), 1);
-                while (index < batchSize) {
-                    Map<Record, CDCImporter> recordChannelMap = new HashMap<>(incrementalRecordMap.size(), 1);
-                    for (Entry<CDCImporter, BlockingQueue<Record>> entry : incrementalRecordMap.entrySet()) {
-                        BlockingQueue<Record> blockingQueue = entry.getValue();
-                        if (null == blockingQueue.peek()) {
-                            continue;
-                        }
-                        Record record = blockingQueue.poll();
-                        if (record instanceof FinishedRecord) {
-                            continue;
-                        }
-                        recordChannelMap.put(record, entry.getKey());
-                        pipelineChannelPositions.computeIfAbsent(entry.getKey(), key -> record);
-                    }
-                    List<DataRecord> filterRecord = recordChannelMap.keySet().stream().filter(each -> each instanceof DataRecord).map(each -> (DataRecord) each).collect(Collectors.toList());
-                    if (filterRecord.isEmpty()) {
+                final Map<CDCImporter, CDCAckPosition> cdcAckPositionMap = new HashMap<>();
+                List<DataRecord> dataRecords = new LinkedList<>();
+                for (int i = 0; i < batchSize; i++) {
+                    DataRecord minimumDataRecord = findMinimumDataRecordAndSavePosition(cdcAckPositionMap);
+                    if (null == minimumDataRecord) {
                         break;
                     }
-                    DataRecord minDataRecord = Collections.min(filterRecord, dataRecordComparator);
-                    dataRecords.add(minDataRecord);
-                    pipelineChannelPositions.put(recordChannelMap.get(minDataRecord), minDataRecord);
-                    index++;
+                    dataRecords.add(minimumDataRecord);
                 }
                 if (dataRecords.isEmpty()) {
                     ThreadUtil.sleep(200, TimeUnit.MILLISECONDS);
                 } else {
-                    writeImmediately(dataRecords, pipelineChannelPositions);
+                    writeImmediately(dataRecords, cdcAckPositionMap);
+                }
+            }
+        }
+        
+        private DataRecord findMinimumDataRecordAndSavePosition(final Map<CDCImporter, CDCAckPosition> cdcAckPositionMap) {
```

Comment:
It needs unit test, if it's not convenient to do unit test, could we extract it as public class